lsp-protocol.el

```diff
@@ -106,22 +106,20 @@ Example usage with `dash`.
                       ((listp object) (-all? (lambda (prop)
                                                (plist-member object prop))
                                              ',required))))
-                  `(cl-defun ,(intern (format "lsp-make-%s" (s-dashed-words (symbol-name interface))))
-                       (&key ,@(-map (-lambda ((key))
-                                       (intern (substring (symbol-name key) 1))) params))
+                  `(defun ,(intern (format "lsp-make-%s" (s-dashed-words (symbol-name interface))))
+                       (&rest plist)
```

Comment:
We're losing parameter quick view via `eldoc` with this. Do you want to do this to passing extra keys?
`&allow-other-keys` may help

Reply:
We cant have both: if they are named we should process them one by one thus we will lose the ability to distinguish the case when the parameter is present or not.

Reply:
I pushed a version which generates a doc which lists what are the expected keys.

Reply:
Can't we define something like this?
``` elisp
(cl-defun lsp--capf-filter-candidates (items
                                       &rest plist
                                       &key lsp-items
                                         &allow-other-keys)
)
```

Reply:
If we agree on that this will be breaking change and should be addressed separately, not in issue which is a blocker for using multiple language servers in one file. This approach also has disadvantages like the fact that it is mandatory to list all non-optional params and with a lot of required elements it will make the code less clear which is much bigger problem than the lack of eldoc for the params.

Reply:
Maybe I'm misunderstanding something here.
Without your new function signature, it turned into `(defun lsp-make-xyz (&rest plist))`, so you can access `plist` directly. However, the eldoc for params will be lost.

What I'm proposed is `(cl-defun lsp-make-xyz (&rest plist &key x y z))`, which will still gave you the same `plist` for using, and the eldoc for params is preserved.
What's the breaking change here?

Reply:
I thought that you are proposing to have initial list of mandatory parameters(at least this is what your sample have). The latest example would work.

Reply:
Sorry for bad example. The first example is something from `lsp-mode.el` so it put it as it.